plasmapy/particles/particle_class.py

```diff
@@ -2555,7 +2558,14 @@ def molecule(
 * **Custom particles**
 
     `~plasmapy.particles.particle_class.CustomParticle` instances are
-    particle-like because particle properties are provided in physical units.
+    particle-like because particle properties are provided in physical
+    units. A `~astropy.units.Quantity` with a physical type of mass or
+    charge is |particle-like| because it can be used to generate a
+    |CustomParticle|.
```

Comment:
Yes, but that's a bit of a crippled particle, no? If you just pass in a charge, you get a massless charge, which is weird. I would prefer not to extend this definition to "whatever works to create a CustomParticle, too" and keep `ParticleLike` simple.

Reply:
The main reason for including `Quantity` here is to be consistent with how `particle_input` currently works. For consistency with type checkers, type hint annotations of `ParticleLike` would often need to become `Union[ParticleLike, u.Quantity]`, which would be inconsistent with the annotation that `particle_input` currently needs.  There are times too (like in `RelativisticBody`) where a mass is all that's needed. 

In #1724, I made an attempt to [decompose `ParticleLike` into different & more specific typing constructs](https://github.com/PlasmaPy/PlasmaPy/blob/645382e400b2bdca54e2b0c15bd01a4cea2c1047/plasmapy/utils/typing.py#L32), where `ParticleLike` would include `CustomParticle`, `Particle`, or `ParticleList` -like arguments, and there would be other typing constructs specific to `Particle`, `CustomParticle`, or `ParticleList` -like arguments. I'm still trying to figure out how to organize the typing constructs, so feedback would be much appreciated.  Would it be okay to continue this discussion in #1724?

I didn't realize this back in 2017 or 2018, but the biggest downside of having lots of flexibility for `Particle` is that the type hint annotations become kind of a pain to deal with! 😮‍💨 Sigh!

Reply:
Absolutely, 1724 it is, then!